### PR TITLE
add unit tests

### PR DESCRIPTION
**Pull request by m00g3n**
**Description**
- add unit tests

**Related issue(s)**
kyma-project/console#684

### CODE REVIEW

File: library/package.json [hunk 1]
```diff
@@ -50,7 +50,7 @@
     "postpack": "rm -rf ./README.md && rm -rf ./LICENSE"
   },
   "dependencies": {
-    "asyncapi-parser": "^0.7.0",
+    "asyncapi-parser": "^0.7.1",
```
Comment (magicmatatjahu) [inline library/package.json]: Already is `0.8.2` version.

File: library/src/helpers/parser.ts [hunk 2]
```diff
@@ -12,25 +10,37 @@ import {
...
+) => Promise<AsyncApi>;
+
+type ParseDocumentFromURL = (
+  url: string,
+  requestOptions?: RequestInit,
+  parserOptions?: ParserOptions,
+) => Promise<AsyncApi>;
+
+export default class Parser {
+  private parseDocument: ParseDocument;
```
Comment (magicmatatjahu) [inline library/src/helpers/parser.ts]: `parse` and `parseFromURL`.

File: library/src/helpers/parser.ts [hunk 3]
```diff
@@ -12,25 +10,37 @@ import {
   AsyncApiProps,
 } from '../types';
 
+import { UNSUPPORTED_SCHEMA_VERSION } from '../constants';
+
 type ParserOptions = AsyncApiProps['parserOptions'];
 
-import { UNSUPPORTED_SCHEMA_VERSION } from '../constants';
+type ParseDocument = (
```
Comment (magicmatatjahu) [inline library/src/helpers/parser.ts]: `ParseSchema` and `ParseSchemaFromURL`

File: library/src/helpers/parser.ts [hunk 4]
```diff
@@ -68,10 +70,7 @@ class Parser {
       return { data: null, error: { message: err.message } };
     }
 
-    if (
-      err.parsedJSON &&
-      !this.isCorrectSchemaVersion(err.parsedJSON.asyncapi)
-    ) {
+    if (err.parsedJSON && err.parsedJSON.asyncapi.startsWith('1')) {
```
Comment (magicmatatjahu) [inline library/src/helpers/parser.ts]: Please revert this to `isCorrectSchemaVersion` function.

Comment (m00g3n) [reply library/src/helpers/parser.ts]: - `isCorrectSchemaVersion` only checks the first character e.g. `-1.2.3` is not a correct schema version but the function would return `true`
- it is oneliner used just in this place

File: library/src/helpers/parser.ts (hunk 2 above)
Comment (m00g3n) [reply library/src/helpers/parser.ts]: This is the tricky one we have a class that already has `parse` method.
I see two solutions:

1. rename class method from `parse` to `do` than rename `parseDocument` to `parse` and `parseDoumentFromURL` to `parseURL`
1. rename from `parseDocument` to `parseFn` (or similar) and `parseDoumentFromUR`to `parseURLFn`

Comment (magicmatatjahu) [reply library/src/helpers/parser.ts]: 3. rename to `parseSchema{FromURL}`

**Pull request merged**